tests/phpunit/tests/functions/wpUniquePrefixedId.php

```diff
@@ -0,0 +1,74 @@
+<?php
+
+/**
+ * Test cases for the `wp_unique_prefixed_id()` function.
+ *
+ * @package WordPress\UnitTests
+ *
+ * @since 6.4
+ *
+ * @group functions.php
+ * @covers ::wp_unique_prefixed_id
+ */
+class Tests_Functions_WpUniquePrefixedId extends WP_UnitTestCase {
+
+	/**
+	 * Tests that the expected unique prefixed IDs are created.
+	 *
+	 * @ticket 59681
+	 *
+	 * @dataProvider data_should_create_unique_prefixed_ids
+	 *
+	 * @param mixed $prefix   The prefix.
+	 * @param array $expected The next two expected IDs.
+	 */
+	public function test_should_create_unique_prefixed_ids( $prefix, $expected ) {
+		$id1 = wp_unique_prefixed_id( $prefix );
+		$id2 = wp_unique_prefixed_id( $prefix );
+
+		$this->assertNotSame( $id1, $id2, 'The IDs are not unique.' );
+		$this->assertSame( $expected, array( $id1, $id2 ), 'The IDs did not match the expected values.' );
+	}
+
+	/**
+	 * Data provider.
+	 *
+	 * @return array[]
+	 */
+	public function data_should_create_unique_prefixed_ids() {
+		return array(
+			'prefix as null'               => array(
```

Comment:
The prefix should never be `null`,  as it's documented to require a `string`. And the return is also a `string`.

What happens if `null` is passed to it?
PHP automatically juggles the key to an empty string https://3v4l.org/NqAUJ.

Thinking though non-string values would not be part of the happy-path coverage and could be misleading or confusing to contributors reading the tests.

Reply:
Note: Separating the non-string values from the data provider is not blocking for the PR. The tests can be iterated upon after commit.

Reply:
cc @costdev